File: LeetCode/Python/validParenthesis.py
def isValid(self, s: str) -> bool:
        if len(s) == 0:
            return True
        
        if len(s) < 2:
            return False
        
        stack = []
        dic = {
            "(": ")",
            "{": "}",
            "[": "]"
        }
        
        for ch in s:
            if ch in dic:
                stack.append(dic[ch])
            else:
                if len(stack) == 0 or ch != stack.pop():
                    return False
        
        return True if len(stack) == 0 else False

File: LeetCode/CSharp/ValidParenthesis.cs
public class Solution {
    public bool IsValid(string s) {
        
        if(s.Length == 0) return true;
        if(s.Length < 2) return false;
        
        var dict = new Dictionary<char, char>() {
            { '(' , ')'},
            { '[' , ']'},
            { '{' , '}'}
        };

        var stack = new Stack<char>();

        for(int i = 0; i < s.Length; i++)
        {
            if(dict.ContainsKey(s[i]))
            {
                stack.Push(dict[s[i]]);
            }
            else
            {   
                if(stack.Count == 0 || s[i] != stack.Pop()) return false;
            }
        }

        if(stack.Count != 0) return false;

        return true;
    }
}